sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsRequestInfo.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * This class represents diagnostic information for transport requests (calls to a replica in direct mode, calls to
+ * the Gateway for example to get metadata like physical addresses of replica for a partition.
+ */
+public final class CosmosDiagnosticsRequestInfo {
+
+    private final String activityId;
+    private final String partitionId;
+    private final String partitionKeyRangeId;
+    private final String requestType;
+    private final Instant startTime;
+    private final Duration duration;
+    private final Duration backendLatency;
+    private final Collection<CosmosDiagnosticsRequestEvent> events;
+
+    CosmosDiagnosticsRequestInfo(
+        String activityId,
+        String partitionId, // nullable
+        String partitionKeyRangeId, // nullable
+        String requestType,
+        Instant startTime,
+        Duration duration, // nullable
+        Duration backendLatency, // nullable
+        double requestCharge,
+        int responsePayloadSizeInBytes,
+        int statusCode,
+        int subStatusCode,
+        Collection<CosmosDiagnosticsRequestEvent> events) {
+
+        checkNotNull(activityId, "Argument 'activityId' must not be null.");
+        checkNotNull(requestType, "Argument 'requestType' must not be null.");
+        checkNotNull(startTime, "Argument 'startTime' must not be null.");
+        checkNotNull(events, "Argument 'events' must not be null.");
```

Comment:
Is it okay to allow empty `events`?

Reply:
yes I think so. like for addressResolution requests, events will be empty